doc/config-generic-update-rollback/SONiC_Generic_Config_Update_and_Rollback_Design.md

```diff
@@ -0,0 +1,617 @@
+```
+For detailed information about the JSON patch operations refer to the  section 4(Operations) of [RFC 6902](https://tools.ietf.org/html/rfc6902).
+
+
+
+**Operation Code**
+
+- replace - Set ConfigDB entry described in Path to be equal to the data specified in Value
+
+- add - Create a new ConfigDB entry described in Path and set it Value
+
+- remove - Delete the ConfigDB entry described in Path
+
+- copy - Copy the ConfigDB entry  specified in the FromPath to create a ConfigDB entry specified in Path
+- move - Copy the ConfigDB entry specified in the FromPath to create a ConfigDB entry specified in Path and then delete the ConfigDB entry in FromPath
+
+
+
+**Path**
+
+Describes the location of the ConfigDB entry which is being processed. The path string can be dissected into below elements.
+
+- Table Name - The ConfigDB table corresponding to the entry
+- Key - The Key string to identify the row data within the ConfigDB table
+- Field - The column_key to identify the ConfigDB entry within the identified row data
+
+e.g  "/VLAN_MEMBER/Vlan10|Ethernet0/tagging_mode"
+
+- Table: VLAN_MEMBER
+- Key: Vlan10|Ethernet0
+- Field: tagging_mode
+
+
+**FromPath**
+
+- Same as Path used in copy and move operations
+
+
+
+**Value**
+
+- Data that is used to patch the ConfigDB entry specified in path
+#### 3.6.1.2 ApplyPatch Metadata
+
+A new file will be added to the system which will contain the `apply-patch` metadata.
+
+The format of the file would be:
+```
+{
+  "tables: {
+    "<TABLE-NAME>": {
+      "upstream-tables": ["<TABLE1>, "<TABLE2>" ...],
+      "services-to-restart": ["<SERVICE1>", "<SERVICE2>" ...],
+    },
+    .
+    .
+    .
+  },
+  "default-delay-between-table-updates": <INTEGER>
+}
+```
+
+The document contain info describing the dependency graph of tables (i.e. the topology), services associated with table, and also the delay duration between different tables updates. The granuality is based on the table, as using just the table names we can re-order, and generate all the need operations
+
+- tables - contains the all the information need for the tables
+- TABLE-NAME - The table name correspndong to atables from ConfigDB, will contain all the tables metadata. There can be multiple different TABLE-NAME keys/dictionaries.
+- upstream-tables - The list of all tables that need to be updated first before current table. There can be multiple upstream tables.
+- services-to-restart - The list of services to restart after updating the current table. There can be multiple services to restart.
+- default-delay-between-table-updates - Some daemons are listening to ConfigDB changes, so artificial sleep to give a pause between two table updates as to ensure that all internal daemons have digested the previous update, before the next one arrives.
+
+### 3.6.2 CLI
+
+#### 3.6.2.1 Configuration Commands
+
+**apply-config**
+*Command Format*
+
+config apply-patch <*patch-filename*> [dry-run] [verbose]
```

Comment:
>config apply-patch [](start = 0, length = 18)

It is good to integrated into existing `config` CLI. Could you make it consistent in this document?